pinot-core/src/main/java/org/apache/pinot/core/plan/maker/InstancePlanMakerImplV2.java

```diff
@@ -140,9 +142,21 @@ public InstancePlanMakerImplV2(QueryExecutorConfig queryExecutorConfig) {
   @Override
   public Plan makeInstancePlan(List<IndexSegment> indexSegments, QueryContext queryContext,
       ExecutorService executorService, long endTimeMs) {
+    boolean prefetch =
```

Comment:
I'd suggest adding an instance config to enable `prefetch, acquire, release`, and make it false by default. If it is false, we skip the column calculation to avoid the overhead because most use cases don't need this feature

Reply:
Added the instance config.
I feel we still need the separate `prefetch` query option, as we wanted finer control on whether we to prefetch or not. Lmk if that's okay